src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,26 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+; TODO: actually infer the bucket type
+(defn infer-bucket-type
+  "infer bucket type from bucket name"
+  [bucket-name]
+  (if (.startsWith bucket-name "gs://") :gs :s3))
```

Comment:
even though it's perfectly fine, to use java methods on strings, there is a dedicated clojure namespace `clojure.string` for string functions, that provides the same functionality in an idiomatic way, e.g. http://clojuredocs.org/clojure.string/starts-with_q